client/src/js/modules/summary/views/summary.vue

```diff
@@ -0,0 +1,1775 @@
+                    }
+                }
+
+            }
+
+            if (this.searchedSamplePrefix != '') {
+                queryParams.push('&'+'sample='+this.searchedSamplePrefix)
+            }
+
+
+            return queryParams.join("")
+
+ 
+        },
+        clearQueryParams() {
+
+            console.log('clear!')
+
+            this.searchedSamplePrefix = ''
+
+            for (var index in this.filters) {
+                if (this.filters[index].textField !== "PROP") {
+                    this.filters[index].selected = ''
+                    this.filters[index].selectedArr = []
+                    this.filters[index].operand = ''
+                    this.filters[index].value = ''
+
+                }
+ 
+            };
+
+            console.log('filters', this.filters)
+            
+        },
+        onPrefixSearch() {
+
+            this.searchedSamplePrefix = this.tempSamplePrefix;
+            this.searchFilterParams();
+
+        },  
+        convertToCSV(data) {
+
+            const result = [
+                // headers
+                Object.keys(data['0']),
+                // values
+                ...Object.values(data).map(item => Object.values(item))
+            ]
+                .reduce((string, item) => {
+                    string += item.join(',') + '\n';
+                    return string;
+            }, '');
+
+            return result;
+        },
+        exportCSV(csv) {
+
+            const anchor = document.createElement('a');
+            anchor.href = 'data:text/csv;charset=utf-8,' + encodeURIComponent(csv);
+            anchor.target = '_blank';
+            anchor.download = 'summary_export.csv';
+            anchor.click();
+
+        },
+        getProcRow(result, field, type) {
+            // return an array for results, also change decimal places for floats
+                if (result[field]) {
+
+                    var dataArray = result[field].split(",");
+
+                    if( type !== 'text') {
+                        if(type == 'decimal4') {
+                            return dataArray.map(function (x) { 
+                                return parseFloat(x).toFixed(4);
+                             });
+                        } else {
+                            return dataArray.map(function (x) { 
+                                return parseFloat(x).toFixed(2);
+                             });
+                        }
+                    };
+                        
+
+                    return dataArray;
+                }
+                else {
+                    return [];
+                };
+        },
+        checkedColumns(index) {
+            // add columns to selected or deselected object for pill display
+            this.summaryParameters[index].checked = !this.summaryParameters[index].checked;
+
+            if (this.summaryParameters[index].checked == false){
+
+            this.deselectedColumns.push(this.summaryParameters[index]);
```

Comment:
Do we need these two arrays? Could in stead each of `summaryParameters` have a `selected` attribute?